builtin/providers/azurerm/data_source_arm_resource_group.go

```diff
@@ -0,0 +1,49 @@
+package azurerm
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceArmResourceGroup() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceArmResourceGroupRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"location": locationForDataSourceSchema(),
+			"tags":     tagsForDataSourceSchema(),
+		},
+	}
+}
+
+func dataSourceArmResourceGroupRead(d *schema.ResourceData, meta interface{}) error {
+	armClient := meta.(*ArmClient)
+
+	resourceGroupName := d.Get("name").(string)
+	location, getLocationOk := d.GetOk("location")
+	resourceId := fmt.Sprintf("/subscriptions/%s/resourceGroups/%s", armClient.subscriptionId, resourceGroupName)
+
+	d.SetId(resourceId)
+
+	if err := resourceArmResourceGroupRead(d, meta); err != nil {
+		return err
+	}
+
+	if getLocationOk {
```

Comment:
In some cases users might care about using a specific region. Being able to specify the location provides a way to make sure the retrieved resource group matches the desired location or stop by an error.